docs/proposals/agent-healthcheck.md

```diff
@@ -0,0 +1,68 @@
+# Agent healthcheck system
+
+- Authors: Xavier Vello
+- Date: 2018-01-11
+- Status: draft
+- [Discussion](https://github.com/DataDog/datadog-agent/pull/1055)
+
+## Overview
+
+Docker containers need to expose a `healthcheck` command orchestrators can use to confirm the readiness of a deployment, and terminate unhealthy containers. This is also implemented in systemd (readiness probe to start dependencies in order, but also watchdog to restart deadlocked processed). This proposal is created to discuss how to implement this system in a pluggable way.
```

Comment:
Just to be clear, this is about how we expose our health check about the health of the NodeAgent and Cluster Agent? Not how we interact with and collect health from containers we monitoring

Reply:
Indeed